modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+		if ( has_filter( 'wp_img_tag_add_adjust' ) ) {
+			add_filter( 'wp_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 20, 3 );
+		} else {
+			add_filter( 'the_content', [ $this, 'filter_content_tags' ], 20 );
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 10, 3 );
+		}
+
+		add_filter( 'attachment_fields_to_edit', [ $this, 'add_dominant_color_media_setting' ], 10, 2 );
+		add_filter( 'attachment_fields_to_save', [ $this, 'save_dominant_color_media_setting' ], 10, 2 );
+	}
+
+	/**
+	 * Add the dominant color metadata to the attachment.
+	 *
+	 * @param array $metadata
+	 * @param int $attachment_id
+	 *
+	 * @return array $metadata
+	 */
+	public function dominant_color_metadata( $metadata, $attachment_id ) {
+
+		$dominant_color = $this->get_dominant_color( $attachment_id );
+
+		if ( ! empty( $dominant_color ) ) {
+			$metadata['dominant_color'] = $dominant_color;
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * Add the dominant color metadata to the attachment.
+	 *
+	 * @param array $metadata
+	 * @param int $attachment_id
+	 *
+	 * @return array $metadata
+	 */
+	public function has_transparency_metadata( $metadata, $attachment_id ) {
+
+		$has_transparency = $this->get_has_transparency( $attachment_id );
+
+		if ( ! empty( $has_transparency ) ) {
+			$metadata['has_transparency'] = $has_transparency;
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * @param $filtered_image
+	 * @param $context
+	 * @param $attachment_id
+	 *
+	 * @return string image tag
+	 */
+	public function tag_add_adjust( $filtered_image, $context, $attachment_id ) {
+
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+		if ( isset( $image_meta['dominant_color'] ) ) {
+			$data  = sprintf( 'data-dominantColor="%s"', $image_meta['dominant_color'] );
+			$style = '';
+			if ( strpos( $filtered_image, 'loading="lazy"' ) !== false && ( isset( $image_meta['show_dominant_color'] ) && 1 === (int) $image_meta['show_dominant_color'] ) ) {
+				$style = ' style="--dominant-color: #' . $image_meta['dominant_color'] . '; " ';
+			}
+
+			$extra_class = '';
+
+			if ( ! isset( $image_meta['has_transparency'] ) ) {
+				$data        .= ' data-hasTransparency="false"';
+			} else {
+				$data .= ' data-hasTransparency="true"';
+				$extra_class = ' has-transparency ';
+			}
+
+			$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+
+			$extra_class    .= ( $this->colorislight( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+			$filtered_image = str_replace( 'class="', 'class="' . $extra_class . ' ', $filtered_image );
+		}
+
+		return $filtered_image;
+	}
+
+
+	/**
+	 * @return void
+	 */
+	public function add_styles() {
+		?>
+		<style>
+			img[data-dominantcolor]:not(.has-transparency) {
+				background-color: var(--dominant-color);
+				background-clip: content-box, padding-box;
+			}
+		</style>
+		<?php
+	}
+
+	public function add_scripts() {
+		?>
+<script>
```

Comment:
@pbearne there is no need for any JavaScript, so please remove the inline one here. With the images being opaque, the presence, or lack of a background color makes no difference.